tools/generate_tables_sql_and_testNames.php

```diff
@@ -1,118 +1,234 @@
+an instrument for the table of each instrument it finds in the 
+{instrument_name}.linst file.
+These sql files are written to the tables_sql/ subdirectory.
+
+Ex cmd without SQL execution:
+php generate_tables_sql_and_testNames.php
+../instruments/[instrument_name].linst --to-sql
+
+Ex cmd with SQL execution:
+php generate_tables_sql_and_testNames.php 
+../instruments/[instrument_name].linst --apply
+USAGE; 
+
+    echo $usage;
+};
+
+$execute_mysql_query = $argv[2] === '--apply';
+$data          = file_get_contents($argv[1]);
+$instruments   = explode("{-@-}", trim($data));
+$error_message = "";
+// Do validation
+if (!is_array($instruments)) {
+    $error_message = "Instrument file could not be processed.";
+} else {
+    $pages = array();
+    $sql_query_statement  = "";
+    $sql_query_table_name = "";
+    $sql_query_columns    = array();
+    // Setup Database object.
+    $factory   = \NDB_Factory::singleton();
+    $db        = $factory->database();
+    foreach ($instruments as $instrument) {
+        $items = explode("\n", trim($instrument));
+        foreach ($items as $item) {
+            $bits = explode("{@}", trim($item));
+            if (preg_match("/Examiner[0-9]*/", $bits[1])) {
+                continue;
+            }
+            switch($bits[0]) {
+                // Generate the CREATE TABLE syntax
             case "table":
-                $tablename = $bits[1];
-
-                $filename="../project/tables_sql/".$bits[1].".sql";
-                $output="CREATE TABLE `$bits[1]` (\n";
-                $output.="`CommentID` varchar(255) NOT NULL default '',
-                          `UserID` varchar(255) default NULL,
-                          `Examiner` varchar(255) default NULL,
-                          `Testdate` timestamp DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
-                          `Data_entry_completion_status` enum('Incomplete','Complete') NOT NULL default 'Incomplete',\n";
-            break;
+                // Remove backticks from table name to prevent errors when 
+                // writing SQL patch file.
+                $sql_query_table_name = str_replace('`', '', $db->escape($bits[1]));
+                $file_name = "../project/tables_sql/".$sql_query_table_name.".sql";
+                $output    = "CREATE TABLE `$sql_query_table_name` (\n";
+                $output   .= "`CommentID` varchar(255) NOT NULL default '',\n"
+                    ."`UserID` varchar(255) default NULL,\n"
+                    ."`Examiner` varchar(255) default NULL,\n"
+                    ."`Testdate` timestamp DEFAULT CURRENT_TIMESTAMP ON "
+                    ."UPDATE CURRENT_TIMESTAMP,\n"
+                    ."`Data_entry_completion_status` "
+                    ."enum('Incomplete','Complete') "
+                    ."NOT NULL default 'Incomplete',\n";
+                $sql_query_statement = "CREATE TABLE `$sql_query_table_name` (\n";
```

Comment:
why are we duplication `$output` in `$sql_query_statement` why not use one to generate the other instead

Reply:
I'm not really positive what the question is about but to build a sql query with preventing an injection it's done with individual variables that are assigned with checking as found here:
http://php.net/manual/en/book.pdo.php
The $output is just to hold what the query as a whole would be and saved to a file because that functionality was requested. I think if you look at my PR with the commit history, the functionality wasn't there and later it was included. Maybe that answers your question.